src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -346,6 +346,54 @@ struct sdm_sensor_info
   } //get()
 };
 
+struct xclbin_slots
+{
+  using result_type = query::xclbin_slots::result_type;
+  using slot_info = query::xclbin_slots::slot_info;
+  using slot_id = query::xclbin_slots::slot_id;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    using tokenizer = boost::tokenizer< boost::char_separator<char> >;
+    std::vector<std::string> xclbin_info;
+    std::string errmsg;
+    auto pdev = get_pcidev(device);
+    pdev->sysfs_get("", "xclbinuuid", errmsg, xclbin_info);
+    if (!errmsg.empty())
+      throw xrt_core::query::sysfs_error(errmsg);
+
+    result_type xclbin_data;
+    // xclbin_uuid e.g.
+    // 0 <uuid_slot_0>
+    // 1 <uuid_slot_1>
+    for (auto& line : xclbin_info) {
+      boost::char_separator<char> sep(" ");
+      tokenizer tokens(line, sep);
+      slot_info data = { };
+
+      if (std::distance(tokens.begin(), tokens.end()) == 1) {
+        tokenizer::iterator tok_it = tokens.begin();
+        data.slot = 0;
+        data.uuid = std::string(*tok_it++);
+        xclbin_data.push_back(std::move(data));
+	break;
+      }
+      else if (std::distance(tokens.begin(), tokens.end()) == 2) {
+        tokenizer::iterator tok_it = tokens.begin();
+        data.slot = std::stoi(std::string(*tok_it++));
+        data.uuid = std::string(*tok_it++);
+
+        xclbin_data.push_back(std::move(data));
+      }
+      else
+        throw xrt_core::query::sysfs_error("xclbinid sysfs node corrupted");
+    }
```

Comment:
The parsing code is not specific to Alveo Linux is it?  
I think it belongs in `core/common/query_requests.cpp`